### PR TITLE
PublicDashboards: Refresh ds plugin supported list

### LINKED ISSUE #74945: Task: change supported and unsupported datasources
- [x] Modify [documentation](https://grafana.com/docs/grafana/latest/dashboards/dashboard-public/#supported-data-sources)
- [x] Delete GH discussion note
- [x] Modify frontend alert related to unsupported datasources

### PR DESCRIPTION
**Pull request by juanicabanas**
**What is this feature?**

After [this PR](https://github.com/grafana/grafana/pull/73708), some datasources don't work and need some modifications. This can be checked [here](https://github.com/grafana/grafana/issues/62374)

This PR updates the documentation and the alert shown in the public dashboard modal when an unsupported datasource plugin is being used in the dashboard

**Why do we need this feature?**

[Add a description of the problem the feature is trying to solve.]

**Who is this feature for?**

[Add information on what kind of user the feature is for.]

**Which issue(s) does this PR fix?**:

Fixes https://github.com/grafana/grafana/issues/74945

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: public/app/features/dashboard/components/ShareModal/SharePublicDashboard/SharePublicDashboardUtils.ts [hunk 1]
```diff
@@ -50,14 +52,20 @@ export const publicDashboardPersisted = (publicDashboard?: PublicDashboard): boo
...
+      const dsType = target?.datasource?.type;
+      if (dsType && supportedDatasources.has(dsType)) {
+        const ds = await getDatasourceSrv().get(target.datasource);
+        if (!(ds instanceof DataSourceWithBackend)) {
+          unsupportedDS.add(dsType);
+        }
+      }
+      if (dsType && !supportedDatasources.has(dsType)) {
+        unsupportedDS.add(dsType);
       }
```
Comment (evictorero) [inline public/app/features/dashboard/components/ShareModal/SharePublicDashboard/SharePublicDashboardUtils.ts:69]:
```suggestion
const dsType = target?.datasource?.type;
if (dsType) {
  if (!supportedDatasources.has(dsType)) {
    unsupportedDS.add(dsType);
  } else {
    const ds = await getDatasourceSrv().get(target.datasource);
    if (!(ds instanceof DataSourceWithBackend)) {
      unsupportedDS.add(dsType);
    }
  }
}
```
Maybe reorganizing the logic like this is better/simpler?

File: docs/sources/dashboards/dashboard-public/index.md [hunk 2]
```diff
@@ -223,6 +215,7 @@ If you've confirmed one of these data sources work with public dashboards, let u
         <li>GitHub</li>
         <li>Google BigQuery</li>
         <li>Grafana for YNAB</li>
+        <li>Graphite</li>
```
Comment (evictorero) [inline docs/sources/dashboards/dashboard-public/index.md:218]: Should we introduce now a new state "not compatible"?  In that case, Graphite should be part of that list with Cloudwatch.

File: public/app/features/dashboard/components/ShareModal/SharePublicDashboard/SharePublicDashboardUtils.ts [hunk 3]
```diff
@@ -50,14 +52,20 @@ export const publicDashboardPersisted = (publicDashboard?: PublicDashboard): boo
...
 
   for (const panel of panels) {
     for (const target of panel.targets) {
-      let ds = target?.datasource?.type;
-      if (ds && !supportedDatasources.has(ds)) {
-        unsupportedDS.add(ds);
+      const dsType = target?.datasource?.type;
+      if (dsType && supportedDatasources.has(dsType)) {
+        const ds = await getDatasourceSrv().get(target.datasource);
+        if (!(ds instanceof DataSourceWithBackend)) {
```
Comment (evictorero) [inline public/app/features/dashboard/components/ShareModal/SharePublicDashboard/SharePublicDashboardUtils.ts:63]: This new logic will not add "Cloudwatch" as unsupported. Should we have a manual list or another source of truth?

**Review by evictorero — COMMENTED** [review]
Left some comments. We can leverage this PR to run the script on [publicdashboards.go](https://github.com/grafana/grafana/blob/main/pkg/services/publicdashboards/publicdashboard.go#L15) and commit the updated list.

File: docs/sources/dashboards/dashboard-public/index.md (hunk 2 above)
Comment (juanicabanas) [reply docs/sources/dashboards/dashboard-public/index.md:218]: mmm yes, makes sense. Currently, we know and we're sure both are not compatible

**Comment by juanicabanas** [comment]
> Left some comments. We can leverage this PR to run the script on [publicdashboards.go](https://github.com/grafana/grafana/blob/main/pkg/services/publicdashboards/publicdashboard.go#L15) and commit the updated list.

The backend-generated list hasn't changed, because there's no way to know if a ds extends from DataSourceWithBackend **or** calls super.query from a backend layer. That's why I was forced to change it in the frontend method.

**Comment by evictorero** [comment]
> > Left some comments. We can leverage this PR to run the script on [publicdashboards.go](https://github.com/grafana/grafana/blob/main/pkg/services/publicdashboards/publicdashboard.go#L15) and commit the updated list.
> 
> The backend-generated list hasn't changed, because there's no way to know if a ds extends from DataSourceWithBackend **or** calls super.query from a backend layer. That's why I was forced to change it in the frontend method.

I know but if you run it you will find some changes, I was wondering if we can add those changes in this PR as well.

File: pkg/services/publicdashboards/commands/generate_datasources/generate.go [hunk 4]
```diff
@@ -11,6 +11,9 @@ import (
 )
 
 var grafanaDatasources = []string{expr.DatasourceType, "datasource"}
+var unsupportedDataSourcesMap = map[string]bool{
```
Comment (evictorero) [inline pkg/services/publicdashboards/commands/generate_datasources/generate.go:14]: Can we add a comment here to explain why we are doing this? This should be removed after this [PR](https://github.com/grafana/grafana/issues/74872) is merged

**Review by evictorero — APPROVED** [review]

**Pull request merged**